lib/aggregate/foreign_key_reference.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Aggregate::ForeignKeyReference
+  attr_reader :id
+
+  def initialize(klass, id_or_value)
+    @class = klass
+
+    if id_or_value.is_a?(ActiveRecord::Base)
+      @value = id_or_value
+      @id = @value.id
+    else
+      @id = id_or_value.to_i
```

Comment:
we're slightly changing the expectation if `id_or_value` is `nil`. Previously, we checked for `nil` and would return `nil`. With this, we will mark the id as `0` and aren't setting `@value`, so accessing the value will cause an `ActiveRecord::RecordNotFound`. Is that expected?